src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SerializerExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using OmniSharp.Extensions.LanguageServer.Protocol.Serialization;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal static class SerializerExtensions
+    {
+        public static void RegisterConverter(this Serializer serializer, Newtonsoft.Json.JsonConverter converter)
+        {
+            serializer.JsonSerializer.Converters.Add(converter);
+            serializer.Settings.Converters.Add(converter);
```

Comment:
Soooo it's weird. Inside csharp-language-server-protocol it seems that sometimes `JsonSerializer` is used and sometimes Serializer.Settings.Converters is used. I can't explain to you exactly how to tell which will be used, but I do know that in this case we need to add it to Settings.Converts. A cursory view of the recent code in this area suggests that this will normalize to always use Serializer.JsonSerializer in newer versions.